app/services/external_payment_processor.rb

```diff
@@ -19,4 +34,8 @@ def self.payment_processor
     StripePaymentProcessor
   end
   
+  def self.create_new_instance_from_response(payment_response)
+    payment_response.try(:id).present? ? new(processed: payment_response) : new(error: payment_response)
```

Comment:
I personally limit using ternary operators, as code is losing readability, I am not really sure also about `payment_response.try(:id).present?` - as I mentioned above - it could just be in a StripePaymentProcessor method `#processed?` which would give a definite statement on whether it is valid or not.